### PR TITLE
fix #325 and #1457

### LINKED ISSUE #325: expression of e.g. "image"
Who hasn't made the following typo: trying to consider the image of a map, say
R=QQ[x]; f=matrix{{x}};
one types
M=image
forgetting f. then immediately corrects it:
i4 : M=image f
but the output is not what is expected:
o4 = { \* Function \* } | x |

why "{ \* Function \* }" instead of "image"? apparently when we redefine M, image loses its hasAttribute(image,ReverseDictionary) and therefore its expression becomes { \* Function \* }.

This is a minor issue, but I'm wondering if it can be easily fixed.

### PR DESCRIPTION
**Pull request by pzinn**
The option `Boxes` of `netList` can now take a two-element list corresponding to which rows/columns are separated by lines, see examples below. This allows for a neater encoding of `net MatrixExpression, MatrixDegreeExpression, VectorExpression` (possibly other `net` routines could benefit from it too) 
```
i1 : t=table(3,6,(i,j)->i*j)

o1 = {{0, 0, 0, 0, 0, 0}, {0, 1, 2, 3, 4, 5}, {0, 2, 4, 6, 8, 10}}

o1 : List

i2 : netList(t,Boxes=>true)

     +-+-+-+-+-+--+
o2 = |0|0|0|0|0|0 |
     +-+-+-+-+-+--+
     |0|1|2|3|4|5 |
     +-+-+-+-+-+--+
     |0|2|4|6|8|10|
     +-+-+-+-+-+--+

i3 : netList(t,Boxes=>{false,true})

o3 = |0|0|0|0|0|0 |
     |0|1|2|3|4|5 |
     |0|2|4|6|8|10|

i4 : netList(t,Boxes=>{{1},{1}},HorizontalSpace=>1)

o4 = 0 | 0 0 0 0 0 
     --+-----------
     0 | 1 2 3 4 5 
     0 | 2 4 6 8 10

i5 : netList(t,Boxes=>{{0,1,3},{0,1,6}},HorizontalSpace=>1)

     +---+------------+
o5 = | 0 | 0 0 0 0 0  |
     +---+------------+
     | 0 | 1 2 3 4 5  |
     | 0 | 2 4 6 8 10 |
     +---+------------+

i6 : netList(t,Boxes=>{true,{0,1,6}},VerticalSpace=>1,HorizontalSpace=>2)

     +-----+------------------+
     |     |                  |
o6 = |  0  |  0  0  0  0  0   |
     |     |                  |
     +-----+------------------+
     |     |                  |
     |  0  |  1  2  3  4  5   |
     |     |                  |
     +-----+------------------+
     |     |                  |
     |  0  |  2  4  6  8  10  |
     |     |                  |
     +-----+------------------+
```

### CODE REVIEW

File: M2/Macaulay2/packages/Macaulay2Doc/doc3.m2 [hunk 1]
```diff
@@ -1053,7 +1053,7 @@ document { Key => {(netList, VisibleList),
      Usage => "netList v",
      Inputs => { 
 	  "v" => {"a list of lists of things to be converted to nets and displayed as a table in a net"},
-	  Boxes => Boolean => {"whether to draw boxes around the individual nets"},
+	  Boxes => {"whether to draw boxes around the individual nets. can be a Boolean or a pair of Booleans/lists of rows/columns to insert lines"},
```
Comment (DanGrayson) [inline M2/Macaulay2/packages/Macaulay2Doc/doc3.m2:1056]: This description is hard to understand.  Can you rephrase it?  Also capitalize the first word of sentences and end sentences with periods.

File: M2/Macaulay2/m2/nets.m2 [hunk 2]
```diff
@@ -220,40 +220,36 @@ alignmentFunctions := new HashTable from {
 netList VisibleList := o -> (x) -> (
```
Comment (DanGrayson) [inline M2/Macaulay2/m2/nets.m2:220]: It would be cool if negative numbers could be used to indicate offsets from the end of the list, i.e., if these two would give the same answer:
```m2

i16 : netList (table(5,5,i->i),HorizontalSpace=>1,Boxes=>{{0,1,3},{5}})

      -----------------------------------+
o16 = (0, 0) (0, 1) (0, 2) (0, 3) (0, 4) |
      -----------------------------------+
      (1, 0) (1, 1) (1, 2) (1, 3) (1, 4) |
      (2, 0) (2, 1) (2, 2) (2, 3) (2, 4) |
      -----------------------------------+
      (3, 0) (3, 1) (3, 2) (3, 3) (3, 4) |
      (4, 0) (4, 1) (4, 2) (4, 3) (4, 4) |

i17 : netList (table(5,5,i->i),HorizontalSpace=>1,Boxes=>{{0,1,3},{-1}})

      ----------------------------------
o17 = (0, 0) (0, 1) (0, 2) (0, 3) (0, 4)
      ----------------------------------
      (1, 0) (1, 1) (1, 2) (1, 3) (1, 4)
      (2, 0) (2, 1) (2, 2) (2, 3) (2, 4)
      ----------------------------------
      (3, 0) (3, 1) (3, 2) (3, 3) (3, 4)
      (4, 0) (4, 1) (4, 2) (4, 3) (4, 4)
```

**Review by DanGrayson — CHANGES_REQUESTED** [review]
Good solution!

```m2

i7 : transpose random (R^1, R^{3:-3})

           |           3     2         2     2                   2       2     3         |
o7 = {-3}  |         3x  + 7x x  + 2x x  + 9x x  + 2x x x  + 7x x  + 8x x  + 6x          |
           |           1     1 2     1 2     1 3     1 2 3     1 3     2 3     3         |
           |                                                                             |
           |         3     2        2     3     2                   2       2     3      |
     {-3}  |       4x  + 6x x  + x x  + 6x  + 3x x  + 3x x x  + 2x x  + 8x x  + 7x       |
           |         1     1 2    1 2     2     1 3     1 2 3     1 3     2 3     3      |
           |                                                                             |
           |    3     2         2     3     2                 2         2       2     3  |
     {-3}  |  4x  + 3x x  + 7x x  + 2x  + 2x x  + 7x x x  + 9x x  + 8x x  + 7x x  + 3x   |
           |    1     1 2     1 2     2     1 3     1 2 3     2 3     1 3     2 3     3  |

             3       1
o7 : Matrix R  <--- R
```

File: M2/Macaulay2/m2/nets.m2 (hunk 2 above)
Comment (pzinn) [reply M2/Macaulay2/m2/nets.m2:220]: I thought about negative numbers. but I don't think it makes much arithmetic sense to have the column *after* the last column labelled as "-1". so I'd rather not.

Comment (DanGrayson) [reply M2/Macaulay2/m2/nets.m2:220]: okay

**Comment by DanGrayson** [comment]
PS: Could you add something to the appropriate entry in `changes.m2`, so people learn about this improvement?
